docs/development/design/future-builder.rst

```diff
@@ -0,0 +1,230 @@
+Future Builder
+==============
+
+.. contents::
+   :local:
+   :depth: 2
+
+This document is a continuation of Santos' work about "`Explicit Builders`_".
+It builds on top of that document some extra features and makes some decisions about the final goal,
+proposing a clear direction to move forward with intermediate steps keeping backward and forward compatibility.
+
+.. _Explicit Builders: https://github.com/readthedocs/readthedocs.org/pull/8103/
+
+
+Goals
+-----
+
+* Keep the current builder working as-is
+* Keep backward and forward (with intermediate steps) compatibility
+* Define a clear support for newbie, intermediate and advanced users
+* Allow users to override a command, run pre/post hook commands or define all commands by themselves
```

Comment:
Note there are people already doing this in hacky ways: https://github.com/NiklasRosenstein/readthedocs-custom-steps